<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainwindow</class>
 <widget class="QWidget" name="mainwindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>774</width>
    <height>719</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mainwindow</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>40</y>
     <width>191</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Введите размер массива:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>80</y>
     <width>181</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Введите массив:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>270</y>
     <width>191</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Отсортированный массив:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>470</y>
     <width>171</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Время работы:</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="arraytextEdit">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>70</y>
     <width>511</width>
     <height>171</height>
    </rect>
   </property>
  </widget>
  <widget class="QTextEdit" name="sortedArraytextEdit">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>270</y>
     <width>511</width>
     <height>171</height>
    </rect>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <widget class="QTextEdit" name="timetextEdit">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>460</y>
     <width>511</width>
     <height>31</height>
    </rect>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>215</y>
     <width>201</width>
     <height>25</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Heap sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Merge sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Quick sort</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Interpolar sort</string>
    </property>
   </item>
  </widget>
  <widget class="QSpinBox" name="sizespinBox">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>30</y>
     <width>511</width>
     <height>26</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>1000</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>520</y>
     <width>181</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Введите число:</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="digitspinBox">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>510</y>
     <width>511</width>
     <height>26</height>
    </rect>
   </property>
   <property name="minimum">
    <number>-2147483647</number>
   </property>
   <property name="maximum">
    <number>2147483647</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>610</y>
     <width>181</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Индекс в массиве:</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="binsearchtextEdit">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>600</y>
     <width>511</width>
     <height>31</height>
    </rect>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>560</y>
     <width>731</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>найти индекс числа в массиве</string>
   </property>
  </widget>
  <widget class="QWidget" name="widget" native="true">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>10</y>
     <width>771</width>
     <height>691</height>
    </rect>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <widget class="QPushButton" name="closeButton">
    <property name="geometry">
     <rect>
      <x>727</x>
      <y>10</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QPushButton" name="nextMoveButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>610</y>
      <width>751</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Следующий шаг</string>
    </property>
   </widget>
   <widget class="QLabel" name="showlabel">
    <property name="geometry">
     <rect>
      <x>235</x>
      <y>10</y>
      <width>301</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Массив отсортирован!</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>480</y>
      <width>131</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Текущий массив:</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="currentArrayplainTextEdit">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>470</y>
      <width>591</width>
      <height>121</height>
     </rect>
    </property>
    <property name="textInteractionFlags">
     <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>731</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>650</y>
     <width>211</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>При использовании binpow:</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="binpowtextEdit">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>640</y>
     <width>511</width>
     <height>31</height>
    </rect>
   </property>
   <property name="textInteractionFlags">
    <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
   </property>
  </widget>
  <zorder>label</zorder>
  <zorder>label_2</zorder>
  <zorder>label_3</zorder>
  <zorder>label_4</zorder>
  <zorder>arraytextEdit</zorder>
  <zorder>sortedArraytextEdit</zorder>
  <zorder>timetextEdit</zorder>
  <zorder>comboBox</zorder>
  <zorder>sizespinBox</zorder>
  <zorder>label_5</zorder>
  <zorder>digitspinBox</zorder>
  <zorder>label_6</zorder>
  <zorder>binsearchtextEdit</zorder>
  <zorder>pushButton</zorder>
  <zorder>label_7</zorder>
  <zorder>binpowtextEdit</zorder>
  <zorder>widget</zorder>
 </widget>
 <resources/>
 <connections/>
</ui>
